### DISCUSSION TITLE
Background image in a PDF file

### DISCUSSION THREAD

**Discussion by IngoManthey** [SixLabors/ImageSharp#1959] [Ideas]
Ich nutze PdfSharpCore. Für die Grafik wird 
ImageSharp genutzt. Kann mir hier jemand helfen, wie ich ein Hintergrundbild einfügen kann?
Danke für jede Hilfe und Tipp.

**Comment by brianpopow** [comment]
Hallo Ingo,

ich versteh leider nicht ganz was genau du erreichen willst. Das klingt ein bisschen so als wär das eher ne Frage für die Leute bei PdfSharpCore, oder?

Was genau hasst du bisher versucht und was hat nicht geklappt? Codebeispiel?

Im übrigen hasst du wahrscheinlich mehr Glück, dass dir jemand antwortet, wenn du dass auf englisch schreiben könntest.

**Reply by IngoManthey** [reply]
@brianpopow 
Thanks for your help. Here my code:
```c#
renderer.PrepareRenderPages();
int pages = renderer.DocumentRenderer.FormattedDocument.PageCount;
for (int i = 1; i <= pages; ++i)
{
    PdfPage page = renderer.PdfDocument.AddPage();

    PageInfo pageInfo = renderer.DocumentRenderer.FormattedDocument.GetPageInfo(i); // A4
    page.Width = pageInfo.Width;
    page.Height = pageInfo.Height;
    page.Orientation = pageInfo.Orientation;
    using XGraphics gfx = XGraphics.FromPdfPage(page);
    gfx.MUH = renderer.Unicode ? PdfFontEncoding.Unicode : PdfFontEncoding.WinAnsi;
    XImage image = XImage.FromFile(Invoice.BackgroundImage);
    gfx.DrawImage(image, page.Width, page.Height);
    renderer.DocumentRenderer.RenderPage(gfx, i);
}
```
Yes, this is actually a question for PdfSharpCore. I've been waiting there for more than 3 months now. Unfortunately nobody answers there. Thank you for your help.

**Reply by JimBobSquarePants** [reply]
I'm afraid we cannot help you here. That `XGraphics` abstraction is an implementation detail of PdfSharpCore and we don't maintain that. 

>I've been waiting there for more than 3 months now. Unfortunately nobody answers there. 

PdfSharpCore is [MIT](https://github.com/ststeiger/PdfSharpCore/blob/145e9ca02db0dc85369b5bbc68460364baa89805/LICENCE.md) licensed so while it is unfortunate from your perspective that they haven't replied they not actually under any obligation to do so and given that you didn't provide an [actual question or expected result](https://github.com/ststeiger/PdfSharpCore/issues/190) when you contacted them I'm not surprised that they haven't replied. 

I would note that you're consuming our code under the [Apache 2.0](https://github.com/SixLabors/ImageSharp/blob/f15bc74a1a42da96e2fd5a1a9e5e5c74ecae3bb3/LICENSE) license which comes with the same lack of obligation so perhaps spamming our discussions forum in German in a channel designed for capturing potential feature requests was not the best move.

I would suggest pulling down their source code and having a look then and if it's a missing feature you can contribute it back to them.

**Reply by IngoManthey** [reply]
Thank you!!!
